presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -332,7 +345,51 @@ private PlanBuilder appendQuantifiedComparisonApplyNode(PlanBuilder subPlan, Qua
             case GREATER_THAN_OR_EQUAL:
                 return planQuantifiedOrderable(subPlan, quantifiedComparison, correlationAllowed);
         }
-        throw new SemanticException(NOT_SUPPORTED, quantifiedComparison, "Quantified comparison with the given quantifier and comparison type is not supported");
+        // all cases are checked, so this exception should never be thrown
+        throw new IllegalStateException("Unexpected combination of quantifier and comparison type: " +
```

Comment:
From the point of view of this method, this should be either an IllegalArgumentException or UnsupportedOperationException. It could happen if the ComparisonType enum is extended in the future.